### PR TITLE
implement the llvm.bswap.* intrinsics

### PR DESCRIPTION
**Pull request by siddharthist**
Progress on #73

### CODE REVIEW

File: what4/src/What4/Interface.hs [hunk 1]
```diff
@@ -1971,6 +1982,64 @@ class (IsExpr (SymExpr sym), HashableF (SymExpr sym)) => IsExprBuilder sym where
...
+-- apply this newtype.
+newtype SymBV' sym w = MkSymBV' { getSymBV :: SymBV sym w }
+
+-- | Join a @Vector@ of smaller bitvectors
+bvJoinVector :: forall sym n w. (1 <= w, IsExprBuilder sym)
+             => sym
+             -> NatRepr w
+             -> Vector.Vector n (SymBV sym w)
+             -> IO (SymBV sym (n * w))
+bvJoinVector sym w = coerce $ Vector.joinWithM @IO @(SymBV' sym) @n bvConcat' w
```
Comment (robdockins) [inline what4/src/What4/Interface.hs]: I think you can just use `getSymBV` here

File: what4/src/What4/Interface.hs [hunk 2]
```diff
@@ -1971,6 +1982,64 @@ class (IsExpr (SymExpr sym), HashableF (SymExpr sym)) => IsExprBuilder sym where
...
+
+-- | Split a bitvector to a @Vector@ of smaller bitvectors
+bvSplitVector :: forall sym n w. (IsExprBuilder sym, 1 <= w, 1 <= n)
+              => sym
+              -> NatRepr n
+              -> NatRepr w
+              -> SymBV sym (n * w)
+              -> IO (Vector.Vector n (SymBV sym w))
+bvSplitVector sym n w x =
+  fmap (getSymBV @sym) <$>
```
Comment (robdockins) [inline what4/src/What4/Interface.hs]: Here is where `coerce` would actually get you something

File: what4/src/What4/Interface.hs (hunk 1 above)
Comment (siddharthist) [reply what4/src/What4/Interface.hs]: That didn't work out.

File: what4/src/What4/Interface.hs (hunk 2 above)
Comment (siddharthist) [reply what4/src/What4/Interface.hs]: Yep, changed!

**Pull request merged**